mlir/include/mlir/Dialect/Tosa/IR/TosaOps.td

```diff
@@ -422,6 +422,8 @@ def Tosa_SigmoidOp : Tosa_ElemWiseUnaryOp<"sigmoid"> {
   let results = (outs
     Tosa_Tensor:$output
   );
+
+  let hasVerifier = 1;
```

Comment:
Instead of adding a manual verifier, would it also work to change
```
class Tosa_ElemWiseUnaryOp<string mnemonic, list<Trait> traits = []> :
    Tosa_Op<mnemonic, !listconcat(traits, [
              DeclareOpInterfaceMethods<InferShapedTypeOpInterface,
                                        ["inferReturnTypeComponents"]>,
              Pure, SameOperandsAndResultElementType])> {
}
```
to say `SameOperandsAndResultType` instead of `SameOperandsAndResultElementType`?

Reply:
The change that you suggested works. But this change affects all unary ops. I will create a new PR for the element wise unary ops with more tests.

Reply:
Sounds good!